test/JDBC/expected/test_conv_float_to_varchar_char-vu-prepare.out

```diff
@@ -0,0 +1,162 @@
+CREATE TABLE float_char_t1 (
+    ID INT IDENTITY(1,1),
+    FloatValue FLOAT,
+    Description VARCHAR(100)
+);
+GO
+-- Create dependent objects for testing
+-- 1. View that uses CAST
+CREATE VIEW float_char_v1
+AS
+SELECT
+    CAST(CAST('-123.456' AS FLOAT)  AS CHAR(30)) AS CastValue
+GO
+
+-- 2. View that uses CONVERT
+CREATE VIEW float_char_v2
+AS
+SELECT 
+    CONVERT(CHAR(30), CAST('-123.456' as FLOAT)) AS ConvertValue
+GO
+
+-- 3. Function using CAST
+CREATE FUNCTION float_char_f1
+(
+    @FloatInput FLOAT
+)
+RETURNS VARCHAR(30)
+AS
+BEGIN
+    RETURN CAST(@FloatInput AS CHAR(30))
+END;
+GO
+
+-- 4. Function using CONVERT
+CREATE FUNCTION float_char_f2
+(
+    @FloatInput FLOAT
+)
+RETURNS VARCHAR(30)
+AS
+BEGIN
+    RETURN CONVERT(CHAR(30), @FloatInput)
+END;
+GO
+
+CREATE PROCEDURE float_char_p1
+    @FloatInput FLOAT
+AS
+BEGIN
+    SELECT 
+        CAST(@FloatInput AS CHAR(30)) AS CastResult,
+        CONVERT(CHAR(30), @FloatInput) AS ConvertResult
+END;
+GO
+
+-- Insert test data
+INSERT INTO float_char_t1 (FloatValue, Description) VALUES
+    (123.456, 'Standard positive decimal'),
+    (-123.456, 'Negative decimal'),
+    (0.0, 'Zero value'),
+    (1E10, 'Scientific notation - large'),
+    (1E-10, 'Scientific notation - small'),
+    (9999999999.99, 'Large decimal'),
+    (0.000000001, 'Very small decimal'),
+	(NULL, 'NULL Value')
+GO
+~~ROW COUNT: 8~~
+
+
+
+CREATE TABLE TestResults (
+    TestID INT IDENTITY(1,1),
+    TestCategory VARCHAR(50),
+    TestName VARCHAR(100),
+    TestScenario VARCHAR(200),
+    ExpectedResult VARCHAR(50),
+    ActualResult VARCHAR(50),
+    TestStatus VARCHAR(20)
+);
+GO
+
+
+CREATE TABLE float_varchar_t1 (
+    ID INT IDENTITY(1,1),
+    FloatValue FLOAT,
+    Description VARCHAR(100)
+);
+GO
+-- Create dependent objects for testing
+-- 1. View that uses CAST
+CREATE VIEW float_varchar_v1
+AS
+SELECT
+    CAST(CAST('-123.456' AS FLOAT)  AS VARCHAR(30)) AS CastValue
+GO
+
+-- 2. View that uses CONVERT
+CREATE VIEW float_varchar_v2
+AS
+SELECT 
+    CONVERT(CHAR(30), CAST('-123.456' as FLOAT)) AS ConvertValue
```

Comment:
Please add test cases for float with some scale like FLOAT(12).

Reply:
Fixed in latest revision